### PR TITLE
make all nodePorts jump to the firewall chain

### PR DESCRIPTION
**Pull request by LexCC**
**What type of PR is this?**
> /kind design
> /kind feature

**What this PR does / why we need it**:
Complete TODO: make all nodePorts jump to the firewall chain
https://github.com/kubernetes/kubernetes/blob/0f4b666469d9bb3d6cb0ef95d7ce9f7930301ec0/pkg/proxy/iptables/proxier.go#L1084
**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
None
```

### CODE REVIEW

File: pkg/proxy/iptables/proxier.go [hunk 1]
```diff
@@ -1081,7 +1081,12 @@ func (proxier *Proxier) syncProxyRules() {
 					// Jump to the service chain.
 					writeLine(proxier.natRules, append(args, "-j", string(svcChain))...)
 				} else {
-					// TODO: Make all nodePorts jump to the firewall chain.
+					for _, lp := range lps {
+						writeLine(proxier.natRules,
```
Comment (thockin) [inline pkg/proxy/iptables/proxier.go]: It's not clear to me why this is ONLY in the non NodeLocal path -- shouldn't it be in both?  The LB IP logic embeds the OnlyLocal decision into the fwChain

**Review by thockin — COMMENTED** [review]
I *think* this can be much simpler, but honestly I am nervous about it.  I had to go examine this logic carefully.  The lack of testing around this makes me nervous -- can you talk about what testing you did to verify this?

**Comment by thockin** [comment]
Assign back to me once it is ready for review

**Comment by danwinship** [comment]
@JieJhih can you respond to Tim's comments?

**Comment by JieJhih** [comment]
> @JieJhih can you respond to Tim's comments?
@thockin @danwinship 
Apologies for the delay in responding to you, I'll work on this PR recently.

**Pull request closed without merging**